lib/data_source/client/github.rb

```diff
@@ -29,30 +42,86 @@ def user_repos
           direction: 'desc',
           per_page: 100
         }
-        with_cache('user_repos') do
-          request('/user/repos', params)
+        responses = with_cache(:user_repos) do
+          all_user_repos = merge_multipage_results('/user/repos', params, 100)
+          all_user_repos
+        end
```

Comment:
The result from the `with_cache` call should be the returned value of the block. So you can just call the method.

```suggestion
        responses = with_cache(:user_repos) do
          merge_multipage_results('/user/repos', params, 100)
        end
```